### ISSUE TITLE
Invalid JSON output with backup command

### ISSUE THREAD

**Issue by florensie** [rustic-rs/rustic#925]
When using `--json` on the backup command I get normal logging mixed with json output. I don't think anyone using `--json` output would have a use for the default logging output. It only gets in the way when trying to use rustic for scripting. Additionally, in some cases this mixed logging completely breaks the json output, see the second to last line below.

```
C:\Users\Florens\test\stuff>rustic backup -r ../repo --host test --json .
using no config file (.\rustic.toml doesn't exist)
enter repository password:
[INFO] repository local:../repo: password is correct.
[INFO] using cache at C:\Users\Florens\AppData\Local\rustic\dafc41415e5541942621e0127f4fb6d6da8ac43445ef7edaf76054ec63f8b420
[00:00:00] reading index...               ████████████████████████████████████████          2/2
[INFO] starting to backup "C:\\Users\\Florens\\test\\stuff"...
[00:00:00] getting latest snapshot...     ████████████████████████████████████████          4/4
[INFO] using parent 6258307e
[00:00:00] backing up...                  ████████████████████████████████████████         6B/6B         869B/s       (ETA 0s)
{
  "time": "2023-11-05T10:58:46.890663100+01:00",
  "program_version": "rustic v0.5.4",
  "parent": "6258307ea84b49867c72cf1284a5703d30fbdf369da5b87533d7db968aa2b0fe",
  "tree": "ecb05d2b6cfb9cdc8635f749138445d3f7ad2bed16ce14ccbaba26e6d7748b68",
  "paths": [
    "C:\\Users\\Florens\\test\\stuff"
  ],
  "hostname": "test",
  "username": "",
  "uid": 0,
  "gid": 0,
  "tags": [],
  "summary": {
    "files_new": 0,
    "files_changed": 0,
    "files_unmodified": 1,
    "dirs_new": 0,
    "dirs_changed": 0,
    "dirs_unmodified": 6,
    "data_blobs": 0,
    "tree_blobs": 0,
    "data_added": 0,
    "data_added_packed": 0,
    "data_added_files": 0,
    "data_added_files_packed": 0,
    "data_added_trees": 0,
    "data_added_trees_packed": 0,
    "total_files_processed": 1,
    "total_dirs_processed": 6,
    "total_bytes_processed": 6,
    "total_dirsize_processed": 940,
    "total_duration": 0.0079027,
    "command": "C:\\Users\\Florens\\scoop\\apps\\rustic\\current\\rustic.exe backup -r ../repo --host test --json .",
    "backup_start": "2023-11-05T10:58:46.894730200+01:00",
    "backup_end": "2023-11-05T10:58:46.898565800+01:00",
    "backup_duration": 0.0038356
  },
  "id": "5b4bc4813b6fbd4efbfe5d5517252826694fd3345b3534c1ca6b16dedc80ecb8"
[INFO] backup of "C:\\Users\\Florens\\test\\stuff" done.
}
```

**Comment by aawsome** [comment]
Thanks for opening this issue. Normally, the logging output should go to stderr while the json output should go to stdout. Something like `rustic ... 2>/dev/null` (or redirecting to a file) should give you the desired json output.
Also, there is `rustic --log-level=off` which completely turns off log messages.

**Comment by florensie** [comment]
I found that `--log-level=off` doesn't completely disable logging either:
```
C:\Users\Florens\test\stuff>rustic backup -r ../repo --host test --log-level off --password "" .
using no config file (.\rustic.toml doesn't exist)
[00:00:00] reading index...               ████████████████████████████████████████          2/2
[00:00:00] getting latest snapshot...     ████████████████████████████████████████          9/9
[00:00:00] backing up...                  ████████████████████████████████████████         6B/6B         879B/s       (ETA 0s)
Files:       0 new, 0 changed, 1 unchanged
Dirs:        0 new, 0 changed, 6 unchanged
Added to the repo: 0 B (raw: 0 B)
processed 1 files, 6 B
snapshot 14808410 successfully saved.
```
`--no-progress` removes a little bit more, but the first line and summary are still printed.

**Comment by florensie** [comment]
> Normally, the logging output should go to stderr while the json output should go to stdout. 

The help page should probably explain this.

**Comment by aawsome** [comment]
> I found that `--log-level=off` doesn't completely disable logging either:
> 
> `--no-progress` removes a little bit more, but the first line and summary are still printed.

The first line is printed to stderr. The problem here is that this can't be a logging output as the logging options may be set in the config file...
The summary is the output of the `backup` command. It is not printed if you specify `--json`.

**Comment by aawsome** [comment]
> > Normally, the logging output should go to stderr while the json output should go to stdout.
> 
> The help page should probably explain this.

See https://rustic.cli.rs/docs/ - the docu is far from good at the moment, but we highly appreciate any help improving it! Note that there is the "edit" symbol on the top right which easily allows to suggest changes!

**Comment by aawsome** [comment]
TL;DR: If you filter out stderr, the output of `rustic backup --json` is valid json. 

A side note about the progress bar: If you are using rustic in an automated script - or to be more precise in a non-terminal output -  (which I think is the main use case for `--json`), no progress bars are printed.

I'm closing this issue as it is the desired behavior.
